near-plugins-derive/src/access_controllable.rs

```diff
@@ -0,0 +1,287 @@
+use crate::access_control_role::new_bitflags_type_ident;
+use crate::utils::{cratename, is_near_bindgen_wrapped_or_marshall};
+use darling::FromMeta;
+use proc_macro::TokenStream;
+use proc_macro2::Span;
+use quote::quote;
+use syn::parse::Parser;
+use syn::{parse_macro_input, AttributeArgs, ItemFn, ItemStruct};
+
+#[derive(Debug, FromMeta)]
+pub struct MacroArgs {
+    #[darling(default)]
+    storage_prefix: Option<String>,
+    role_type: syn::Path,
+}
+
+const DEFAULT_STORAGE_PREFIX: &str = "__acl";
+const DEFAULT_ACL_FIELD_NAME: &str = "__acl";
+const DEFAULT_ACL_TYPE_NAME: &str = "__Acl";
+
+const ERR_PARSE_BITFLAG: &str = "Value does not correspond to a permission";
+const ERR_PARSE_ROLE: &str = "Value does not correspond to a role";
+
+pub fn access_controllable(attrs: TokenStream, item: TokenStream) -> TokenStream {
+    let cratename = cratename();
+    let attr_args = parse_macro_input!(attrs as AttributeArgs);
+    let mut input: ItemStruct = parse_macro_input!(item);
+    let acl_field = syn::Ident::new(DEFAULT_ACL_FIELD_NAME, Span::call_site());
+    let acl_type = syn::Ident::new(DEFAULT_ACL_TYPE_NAME, Span::call_site());
+    let bitflags_type = new_bitflags_type_ident(Span::call_site());
+    if let Err(e) = inject_acl_field(&mut input, &acl_field, &acl_type) {
+        return TokenStream::from(e.to_compile_error());
+    }
+    let ItemStruct { ident, .. } = input.clone();
+
+    let macro_args = match MacroArgs::from_list(&attr_args) {
+        Ok(args) => args,
+        Err(e) => {
+            return TokenStream::from(e.write_errors());
```

Comment:
Why using `write_errors` versus `to_compile_error`?

Reply:
It's a `darling::Error` which doesn't have `to_compile_error()`. [`to_compile_error`](https://docs.rs/syn/1.0.99/syn/struct.Error.html#method.to_compile_error) is a method of `syn::Error`.

The approach using `write_error()` is [recommended](https://github.com/TedDriggs/darling#macro-code).